### PR TITLE
react-select: bump version to 3.0.8

### PR DESCRIPTION
**Pull request by tskarhed**
Prerequisite for #21610

- Changed imports
- Changed the way styles are applied to some react-select components
- Changed name of `Component` -> `ReactSelectComponent`. With the name being so generic it was very hard to understand where it actually came from.
- Updated Snapshots. I took a quick look at TeamPicker and UserPicker, and they seem fine.

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx [hunk 1]
```diff
@@ -233,27 +235,62 @@ export function SelectBase<T>({
     deprecationWarning('Select', 'width property', 'size or className');
     widthClass = 'width-' + width;
   }
+  console.log(size);
+  console.log(inputSizes()[size]);
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx]: Oops

File: packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { PureComponent } from 'react';
 import { css, cx } from 'emotion';
 import { GrafanaTheme } from '@grafana/data';
-import { GroupProps } from 'react-select/lib/components/Group';
+import { GroupProps } from 'react-select/src/components/Group';
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx]: Not 'react-select'?

File: packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
 import React from 'react';
 import { SelectableValue, deprecationWarning } from '@grafana/data';
 // @ts-ignore
-import { default as ReactSelect, Creatable } from '@torkelo/react-select';
+import { default as ReactSelect } from '@torkelo/react-select';
 // @ts-ignore
-import { default as ReactAsyncSelect } from '@torkelo/react-select/lib/Async';
+import Creatable from '@torkelo/react-select/creatable';
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx]: merge with import above?

**Review by peterholmberg — CHANGES_REQUESTED** [review]
There's a mix between imports from torkels fork and the official package here, which is the correct one?

File: packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx (hunk 2 above)
Comment (dehrax) [reply packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx]: `react-select` is used for types, `@types/react-select`

File: packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx (hunk 3 above)
Comment (dehrax) [reply packages/grafana-ui/src/components/Forms/Select/SelectBase.tsx]: https://github.com/JedWatson/react-select/issues/3585

Apparently it is not supported.

File: packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx (hunk 2 above)
Comment (peterholmberg) [reply packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx]:
```suggestion
import { GroupProps } from 'react-select';
```

File: packages/grafana-ui/src/components/Select/SelectOption.test.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
 import SelectOption from './SelectOption';
-import { OptionProps } from 'react-select/lib/components/Option';
+import { OptionProps } from 'react-select/src/components/Option';
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Select/SelectOption.test.tsx]:
```suggestion
import { OptionProps } from 'react-select';
```

File: packages/grafana-ui/src/components/Select/SelectOption.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 // Ignoring because I couldn't get @types/react-select work wih Torkel's fork
 // @ts-ignore
 import { components } from '@torkelo/react-select';
-import { OptionProps } from 'react-select/lib/components/Option';
+import { OptionProps } from 'react-select/src/components/Option';
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Select/SelectOption.tsx]:
```suggestion
import { OptionProps } from 'react-select';
```

File: public/app/core/components/TagFilter/TagOption.tsx [hunk 6]
```diff
@@ -2,7 +2,7 @@
 import React from 'react';
 // @ts-ignore
 import { components } from '@torkelo/react-select';
-import { OptionProps } from 'react-select/lib/components/Option';
+import { OptionProps } from 'react-select/src/components/Option';
```
Comment (peterholmberg) [inline public/app/core/components/TagFilter/TagOption.tsx]:
```suggestion
import { OptionProps } from 'react-select';
```

**Review by peterholmberg — CHANGES_REQUESTED** [review]
Some more import nits

File: packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx (hunk 2 above)
Comment (dehrax) [reply packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx]: I tired it, it doesn't export GroupProps :/

**Comment by dehrax** [comment]
Verifying that old select component works:
- HTTP settings :heavy_check_mark: 
- Cloudwatch query editor :heavy_check_mark:
- FieldDisplayEditor :heavy_check_mark:
- DataSourcePicker in Explore :heavy_check_mark: 
- UnitPicker :heavy_check_mark:

**Comment by dehrax** [comment]
It has nothing to do with this PR since the issue is in master. When there are no options, the dropdown freezes. 
Screenshot from 2020-01-24 09-25-35
There is the bigger issue what causes them to have no data too.

**Comment by tskarhed** [comment]
@peterholmberg Unfortunately the requested changes do not work. It would've been nice with those kind of imports :/

**Comment by dprokop** [comment]
> It has nothing to do with this PR since the issue is in master. When there are no options, the dropdown freezes.

@tskarhed can you report a bug on this?

File: packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx (hunk 2 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Forms/Select/SelectOptionGroup.tsx]: @tskarhed would be good to try figuring a way to use @types/react-select with Torkel's fork

**Review by dprokop — APPROVED** [review]

**Review by peterholmberg — APPROVED** [review]

**Pull request merged**